lib/backends/webgl/glsl-coordinate-lib.ts

```diff
@@ -1216,4 +1224,27 @@ export class CoordsGlslLib extends GlslLib {
         }
         `;
   }
+
+  /**
+   * Produces a packed value getter function for the name and rank given
+   * If a transpose is set proper offsetToCoords mapping will be used
+   * @param name name of the function
+   * @param rank rank of the input
+   * @param transpose whether or not should generate a transpose variation
+   */
+  protected getPackedValueFrom(varName: string, rank: number, width: number, height: number, transpose: boolean):
```

Comment:
>getPackedValueFrom [](start = 12, length = 18)

Do we need this function? Or can we use existing function in this library.

Reply:
Do you mean getA()? the diff beteen getA() and A_Pack() is A_Pack() takes an array as inputs.

Reply:
Do you need an array as function input? For example, for pack, we pass in 1-4 parameters to getA() depending on the input ranks. I think that's also the case here?

---
In reply to: [613794313](https://github.com/microsoft/onnxjs/pull/274#discussion_r613794313) [](ancestors = 613794313)

Reply:
Not entirely impossible to use getA(). it will need a bit clunky code to deal following code:
 for (int k=0; k<((${sharedDim}+1)/2); ++k) {
              a[${arank - 1}] = k;
              b[${brank - 2}] = k;
              value += _A_Pack(a).rrbb * _B_Pack(b).rgrg;
              value += _A_Pack(a).ggaa * _B_Pack(b).baba;
  }
  
It's not quite the same as Pack. For pack, there are four cases for last two dims, but here it's more complicated. Make sense?